src/graph/entities/node.c

```diff
@@ -35,6 +37,33 @@ int Node_IncomeDegree(const Node *n) {
 	return Vector_Size(n->incoming_edges);
 }
 
+void Node_SetLabelID(Node *n, int labelID) {
+	assert(n);
+	n->labelID = labelID;
+}
+
+GrB_Matrix Node_GetMatrix(Node *n) {
+	/* Node's label must be set, 
+	 * otherwise it doesn't make sense to refer to a matrix. */
+	assert(n && n->label);
+
+	// Retrieve matrix from graph if edge matrix isn't set.
+    if(!n->mat) {
+        GraphContext *gc = GraphContext_GetFromTLS();
+        Graph *g = gc->g;
+
+        // Get label matrix.
+        if(n->labelID == GRAPH_UNKNOWN_LABEL) {
+			// Use a zeroed matrix, LEAK!
```

Comment:
These zero-matrix leaks are such a nuisance. At minimum, we should be able to use a static variable to stand in for all of them, but even better would be to identify and return from queries that call for them early. At least until we have features like OPTIONAL MATCH, I think the only query that can produce results and build a zero matrix would use the additive semiring: `()-[:real_reltype|:fake_reltype]->()`.